### DISCUSSION TITLE
How to use Conductor with Standalone Redis cache which has timeout for idle connections.

### DISCUSSION THREAD

**Discussion by vamsibandarupalli** [Netflix/conductor#2388] [Q&A]
I was trying to use conductor with Azure Cache for Redis, which has 10 minutes timeout for idle connections. Conductor was priming 31 connections when started the server and at some point if there are no (read/write)operations performed to redis, Azure is terminating the idle connections as a result getting the broken pipe error.

Also in Conductor [documentation](https://netflix.github.io/conductor/server/#using-standalone-redis-elasticache) it is mentioned maxConnectionPerHost as 31 is there any reason it should be 31? If we decrease it will there be any decrease in performance?

**Comment by aravindanr** [comment]
I am not sure how the timeout for idle connections and the `maxConnectionPerHost` are connected. The error you described is related to connections timing out and `maxConnectionPerHost` controls the number of connections per host.

**Reply by vamsibandarupalli** [reply]
Hi, those are two different questions, not connected.

1. How to use Conductor with Redis Cache which has timeout for idle connections.
2. Is there any specific reason for maxConnectionPerHost to be 31? Or is it just a random number added in the docs [here](https://netflix.github.io/conductor/server/#using-standalone-redis-elasticache)

**Reply by aravindanr** [reply]
Understood.
1. Conductor does not expose a property to modify connection timeout values for Redis. But [RedisClusterConfiguration](https://github.com/Netflix/conductor/blob/main/redis-persistence/src/main/java/com/netflix/conductor/redis/config/RedisClusterConfiguration.java) can be modified to include a timeout property. If you are inclined, please create a PR.
2. It's the latter. It's a number that we internally used at some point and not a mandate. The default value is `10` and it could be set to any desired value.
